draft-steele-cose-merkle-tree-proofs.md

```diff
@@ -60,31 +60,28 @@ informative:
 
 --- abstract
 
-This specification describes three CBOR data structures for primary use in COSE envelopes.
-A CBOR encoding of Merkle Roots for use in COSE payloads.
-A CBOR encoding of Inclusions Proofs for use in COSE unprotected headers.
-A CBOR encoding of Consistency Proofs for use in COSE unprotected headers.
+This specification describes verifiable data structures and associated proof types for use with COSE.
+The extensibility of the approach is demonstrated by providing CBOR encodings for RFC9162.
 
 --- middle
 
 # Introduction
 
-Merkle trees are verifiable data structures that support secure data storage,
+Merkle trees are one of many verifiable data structures that enable tamper evident secure information storage,
 through their ability to protect the integrity of batches of documents or collections of statements.
 
-A merkle proof is a path from a leaf to a root in a merkle tree.
+Merkle trees can be constructed from simple operations such as concatenation and digest via a cryptographic hash function,
+however, more advanced constructions enable proofs of different properties of the underlying verifiable data structure.
 
-Merkle proofs can be used to prove a document is in a database (proof of inclusion),
-or that a smaller set of statements are contained in a large set of statements (selective disclosure proofs).
+Verifiable data structure proofs can be used to prove a document is in a database (proof of inclusion),
+that a database is append only (proof of consistency), that a smaller set of statements are contained
```

Comment:
Are consistency proofs on their own sufficent to show a database has been managed in an append only manner ?

Reply:
The ones from rfc9162 depend on inclusion proofs, but together they prove the append only property. The details are in rfc9162, there is a tree diagram that helps explain this better than the text in our draft.

Reply:
Here is an implementation that might be a bit easier to understand:

https://github.com/transmute-industries/rfc9162/blob/main/test/consistencyProof.test.ts

its base on this https://datatracker.ietf.org/doc/html/rfc9162#section-2.1.5